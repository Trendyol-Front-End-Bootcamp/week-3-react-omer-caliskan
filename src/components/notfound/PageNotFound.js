import React from 'react';

function PageNotFound(){
    return(
        <div className="notfound">
        <img src="https://staticdelivery.nexusmods.com/mods/1151/images/528-0-1447526230.png" />
        <div>
         <h1>NO RESULT</h1>
        <h2>The page you are trying to search has been moved to another universe.</h2>
        </div>
        </div>
    )
}

export default PageNotFound;