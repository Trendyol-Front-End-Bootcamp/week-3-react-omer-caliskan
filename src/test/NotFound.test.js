import React from 'react'
import Enzyme, { shallow } from 'enzyme';
import Adapter from "enzyme-adapter-react-16";
import PageNotFound from "../components/notfound/PageNotFound";

Enzyme.configure({ adapter: new Adapter() })

describe("Not Found Page", () => {
            let notfound

            beforeEach(() => {
                notfound = shallow(<PageNotFound />)
            })

            it("should check page not found class", () => {
                const checkClass = notfound.hasClass("notfound")
        
                expect(checkClass).toBeTruthy()
              })

            it("should render Page Not Found", () => {
                expect(notfound).toMatchSnapshot()
            })

            it("should have right Page Not Found message", () => {
                const message = notfound.find('h2').text()

                expect(message).toEqual("The page you are trying to search has been moved to another universe.")
            })
          })