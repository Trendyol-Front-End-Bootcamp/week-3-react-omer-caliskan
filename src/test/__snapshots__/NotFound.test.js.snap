// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not Found Page should render Page Not Found 1`] = `
"<div className=\\"notfound\\">
  <img src={{...}} alt=\\"Page Not Found\\" />
  <div>
    <h1>
      NO RESULT
    </h1>
    <h2>
      The page you are trying to search has been moved to another universe.
    </h2>
  </div>
</div>"
`;
